### DISCUSSION TITLE
Use local storage instead of session storage?

### DISCUSSION THREAD

**Discussion by Max-Eriksson** [ActiveLogin/ActiveLogin.Authentication#440] [Q&A] [answered]
Is it possible to use local storage instead of session storage, i.e override the behavior in the following file https://github.com/ActiveLogin/ActiveLogin.Authentication/blob/3a3b18a491600d188e85e1863a9d0828b19763da/src/ActiveLogin.Authentication.BankId.AspNetCore/Client/activelogin-main.ts#L30

We're currently migrating from ActiveLogin version 4 and IdentityServer4 to the latest version and in our current version we had to fork the ActiveLogin.Authentication.BankId.AspNetCore and configure the use of local storage instead of session storage. If I remember it correct this fix was needed for login from native anroid apps (react-native) using Azure AD B2C and BankID.

**Comment by PeterOrneholm** [comment, marked as answer]
Hi Max!
Good question. There have always been the opportunity to customize the script, without "forking" the whole library.
The script is "linked" from a separate view, and all views can be overridden, see the docs here:
https://github.com/ActiveLogin/ActiveLogin.Authentication/blob/main/docs/articles/bankid.md#customize-the-ui

Would that fit your needs?

We won't change the default implementation from sessionStorage to localStorage, but you are free to override that if it fits your needs :) By default we don't want that info to be "long lived", but rather to be cleared when the "session" ends (user navigates away/closes browser.
